addons/mail/static/src/messaging/entity_manager.js

```diff
@@ -114,7 +104,7 @@ class EntityManager {
      */
     create(Entity, data = {}) {
         const entity = new Entity({ valid: true });
-        entity.env = this.env;
+        Object.defineProperty(entity, 'env', { get: () => this.env });
```

Comment:
Conceptually I think the env of an entity instance would be better redirected to the env of its entity class?